DRC badge push to main fails on repos with branch protection

After #58 changed the DRC badge update flow to push directly to \`main\` instead of opening a PR, the push fails on repos with branch protection enabled.

## Symptom

DRC workflow runs fail at the badge push step:

```
! [remote rejected] HEAD -> main (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/doplaydo/ligentec-an800'
##[error]Process completed with exit code 1.
```

Example failing run: https://github.com/doplaydo/ligentec-an800/actions/runs/24353269626

## Root cause

#58 replaced:
```bash
git checkout -b update-drc-badge
git push -f origin update-drc-badge
gh pr create ...
```

with:
```bash
git commit -m "update DRC badge [skip ci]"
git push origin HEAD:main
```

The new direct push to \`main\` is rejected by branch protection rules.

## Possible fixes

- Use a PAT or GitHub App token that bypasses branch protection
- Revert to the PR-creation flow
- Push to a side branch and let CI auto-merge

cc @joamatab

Not sure if this is different https://github.com/gdsfactory/quantum-rf-pdk/actions/runs/24355539141/job/71121371479. This happens on a branch that is ahead of `main`, not behind (https://github.com/gdsfactory/quantum-rf-pdk/tree/refs/heads/add-drc).


> Run errors=$(grep -c '<item>' build/lyrdb/all_cells.lyrdb) || errors=0
[add-drc 984937f] update DRC badge [skip ci]
 1 file changed, 15 insertions(+)
 create mode 100644 badges/drc.svg
To https://github.com/gdsfactory/quantum-rf-pdk
 ! [rejected]        HEAD -> main (fetch first)
error: failed to push some refs to 'https://github.com/gdsfactory/quantum-rf-pdk'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Error: Process completed with exit code 1.

--- a/.github/workflows/drc.yml
+++ b/.github/workflows/drc.yml
@@ -69,7 +69,7 @@ jobs:
           if ! git diff --staged --quiet; then
             branch="update-drc-badge"
             git checkout -b "$branch"
-            git commit -m "update DRC badge"
+            git commit -m "update DRC badge [skip ci]"
             git push -f origin "$branch"
             gh pr create --title "update DRC badge" --body "Automated DRC badge update." --base main --head "$branch" || true
           fi

--- a/.github/workflows/drc.yml
+++ b/.github/workflows/drc.yml
@@ -67,6 +67,11 @@ jobs:
           git config user.email "github-actions[bot]@users.noreply.github.com"
           git add badges/drc.svg
           if ! git diff --staged --quiet; then
-            git commit -m "update DRC badge [skip ci]"
-            git push origin HEAD:main
+            branch="update-drc-badge"
+            git checkout -b "$branch"
+            git commit -m "update DRC badge"
+            git push -f origin "$branch"
+            gh pr create --title "update DRC badge" --body "Automated DRC badge update." --base main --head "$branch" || true
           fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}

